### PR TITLE
UI/15496569 search

### PR DESCRIPTION
**Pull request by CzarekDryl**
Added search input, load more button.

https://tw.auroracreation.com/app/tasks/15496569

image

### CODE REVIEW

File: src/components/frame/v5/pages/MembersPage/hooks.ts [hunk 1]
```diff
@@ -27,16 +28,33 @@ export const useMembersPage = () => {
     },
   });
 
+  const searchMembers = (members: Member[]) =>
+    members.filter((member) => {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/v5/pages/MembersPage/hooks.ts:32]: Please, destructure the object

File: src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import debounce from 'lodash.debounce';
+
+import Icon from '~shared/Icon';
+import styles from './SearchInput.module.css';
+import { useSearchContext } from '~context/SearchContext';
+import { useMobile } from '~hooks';
+
+import { SearchInputProps } from './types';
+
+const SearchInput: FC<SearchInputProps> = ({ onSearchButtonClick }) => {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx:12]: Please add displayName value

File: src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import debounce from 'lodash.debounce';
+
+import Icon from '~shared/Icon';
+import styles from './SearchInput.module.css';
+import { useSearchContext } from '~context/SearchContext';
+import { useMobile } from '~hooks';
+
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx:9]: Not needed blank space

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 4]
```diff
@@ -62,13 +65,18 @@ const MembersList: FC<MembersListProps> = ({
...
-        <div className="w-full flex justify-center mt-6">
+      <div className="w-full flex justify-center mt-6">
+        {listLength > 8 && !searchValue && (
           <Link className="text-3" to={viewMoreUrl}>
             {formatMessage({ id: 'viewMore' })}
           </Link>
-        </div>
-      )}
+        )}
+        {listLength > 8 && searchValue && (
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/MembersList/MembersList.tsx:74]: maybe add this to const `src/constants/index.ts`

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

File: src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  const [text, setText] = useState('');
+  const isMobile = useMobile();
+
+  const debounced = useMemo(
+    () => debounce(setSearchValue, 500),
+    [setSearchValue],
+  );
+
+  const onInput: React.ChangeEventHandler<HTMLInputElement> = useCallback(
+    (e) => {
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/Filter/partials/SearchInput/SearchInput.tsx:25]: add type

File: src/components/v5/common/MembersList/hooks.ts [hunk 6]
```diff
@@ -1,14 +1,40 @@
...
   const { name } = colony || {};
   const colonyURL = `${window.location.origin}/colony/${name}`;
 
   const { handleClipboardCopy } = useCopyToClipboard(colonyURL);
 
+  const [startIndex, setStartIndex] = useState(0);
+  const [visibleMembers, setVisibleMembers] = useState<Member[]>([]);
+
+  const listLength = list.length;
+  const membersLimit = isHomePage ? 8 : 12;
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/hooks.ts:17]: please assign values to const

File: src/components/v5/common/MembersList/hooks.ts [hunk 7]
```diff
@@ -1,14 +1,40 @@
...
+  const endIndex = Math.min(startIndex + membersLimit, listLength);
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/hooks.ts:18]: please use useMemo

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**